### PR TITLE
fix: Add sonar analysis to dhis-web

### PR DESCRIPTION
**Pull request by jbee**
First attempt to add sonar analysis for `dhis-web` project.

### CODE REVIEW

**Review by netroms — APPROVED** [review]

File: .github/workflows/analyse-pr.yml [hunk 1]
```diff
@@ -53,6 +53,18 @@ jobs:
...
+        env:
+          BASE_BRANCH: ${{ github.base_ref }}
+          BRANCH: ${{ github.ref }}
+          PR: ${{ github.event.number }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        if: github.event_name == 'pull_request'
+        run: |
+          mvn -f dhis-2/dhis-web/pom.xml clean install -Psonarqube -Pjdk11
+          mvn -f dhis-2/dhis-web/pom.xml sonar:sonar -Dsonar.scm.revision=${{ github.event.pull_request.head.sha }} -Dsonar.projectKey=dhis2_dhis2-core -Dsonar.coverage.jacoco.xmlReportPaths=$(find "$(pwd)" -path '*jacoco.xml' | sed 's/.*/&/' | tr '\n' ',')
```
Comment (netroms) [inline .github/workflows/analyse-pr.yml:66]: Maybe we should use another project key on sonar, Dsonar.projectKey=dhis2_dhis2-core  -> dhis2_dhis2-web?

Comment (jbee) [reply .github/workflows/analyse-pr.yml:66]: Oh yes, I guess this project has to be created manually? @vilkg

Comment (vilkg) [reply .github/workflows/analyse-pr.yml:66]: The project should be created when you do the first analysis. But wouldn't it be better to have everything in one project?

**Comment by vilkg** [comment]
Nice! This workflow is repeating the one we have when running unit and integration tests, so might be worth looking into merging those two to reduce the time it takes to run all the checks.

**Comment by jbee** [comment]
@vilkg I created this in the assumption that a separate configuration and project would be needed. If we can have just one I'd say we rather focus on extending the existing one. I just wanted to get the ball rolling.

**Comment by larshelge** [comment]
Should we close this PR or fix it up?

**Pull request closed without merging**